### PR TITLE
Improve ease of use for QUEST 2023

### LINKED ISSUE #54: Simplify process to uncheck columns for QUEST 2023
From @llhouse2 
> when a teacher chooses variables to visualize they have to UNclick about 15 variables; e.g., seconds and ALL lat -lon variables, Is there a way to, by default, click all variables EXCEPT the one's likely not to include?
I would like the following variables to NOT be selected.
sat_Lat-Center
sat_Lon-Center
sat_Lat-NW
sat_Lon-NW
sat_Lat-SE
sat_Lon-SE
land_Lat-Center
land_Lon-Center
land_Lat-NW
land_Lon-NW
land_Lat-SE
land_Lon-SE
sat_in
land_in
...
Unclick Seconds too

### PR DESCRIPTION
**Pull request by johnbradley**
Changes fetching observations logic to sort INaturalist observations to show the most recent observation.
Unchecks specific columns to simplify use in QUEST 2023.
Changes number of columns displayed on each row from 10 to 6 to make the screen easier to read.

Fixes #54

### CODE REVIEW

**Comment by hlapp** [comment]
Perhaps allow specifying some wildcard pattern to match columns to uncheck or check ("switch") on request. For example, `sat_lon.*` and `sat_lat.*`.

**Comment by thompsonmj** [comment]
It might look better if the columns were visually grouped with one another based on type. Something like ...
**Columns**
* Image Metadata
  ◼ Seconds ◼ Lat ◼ Long
* Satellite Image Coordinates
  ◼ sat_Lat-Center ◼ sat_Lon-Center ◼ sat_Lat-NW ◼ sat_Lon-NW ◼ sat_Lat-SE ◼ sat_Lon_SE
* Satellite Image Colors
  ◼ sat_medianRed ◼ ...
* Land Use Region Proportions
  ◼ GRASSY_LOCAL ◼ ...

**Comment by johnbradley** [comment]
> Perhaps allow specifying some wildcard pattern to match columns to uncheck or check ("switch") on request. For example, sat_lon.* and sat_lat.*.

I'm not quite sure how this would look, but that it seems like quite a few patterns for a user to type into the box:
`sat_lon.*`, `sat_lat.*`, `land_lon.*`, `land_lan.*` `*._in`, then uncheck `Seconds`.

The current list of columns to uncheck are here: https://github.com/Imageomics/Andromeda/blob/57216edea74087d582ed361a6d47c5e3b8b9cb4d/andromeda-ui/backend/parseCSV.ts#L4-L18

**Comment by johnbradley** [comment]
I'm wondering if I should add a button that will uncheck those QUEST "bad" columns. Or perhaps there could be some kind of convention in the CSV for columns that could hint that we should uncheck the column. Or perhaps I am overthinking this.

**Comment by egrace479** [comment]
> The current list of columns to uncheck are here:
> 
> https://github.com/Imageomics/Andromeda/blob/57216edea74087d582ed361a6d47c5e3b8b9cb4d/andromeda-ui/backend/parseCSV.ts#L4-L18

Aren't there also 2 distance from centroid columns that should be unchecked?

**Comment by egrace479** [comment]
Is there a way to do the initial sort by the checked columns, so that all the unchecked columns are at the end?

**Comment by hlapp** [comment]
> I'm not quite sure how this would look, but that it seems like quite a few patterns for a user to type into the box: `sat_lon.*`, `sat_lat.*`, `land_lon.*`, `land_lan.*` `*._in`, then uncheck `Seconds`.

Well, one could use `(sat|land)_(lat|lon|in).*` to capture 5 of the six, but I agree, that's more than simply a "wildcard" and would require some knowledge of regular expression language that most in the target audience likely won't have.

I agree allowing some externally kept list of "columns to hide" that would be custom to a specific deployment is perhaps for now the best way to deal with this. I.e., don't burden the user with specifying this, but make it something a deployment admin does.

**Comment by johnbradley** [comment]
Based on the above feedback I'm going to 
1) Add a config file that lists what I am calling "ancillary columns". An admin could replace this config file with another to customize which column names are considered ancillary.
2) Working of @thompsonmj and @egrace479 's suggestion I will visually group the columns based on their type "regular" or "ancillary".

**Comment by johnbradley** [comment]
I finished updating the code to group column checkboxes based on a config file.

**Review by thompsonmj — APPROVED** [review]

File: andromeda/columnConfig.json
Comment (egrace479) [inline andromeda/columnConfig.json:1]: Isn't the distance from the centroid also supposed to be an ancillary column?

Comment (johnbradley) [reply andromeda/columnConfig.json:1]: I heard back from Leanna that we should include these two columns in the ancillary columns.

**Pull request merged**